### PR TITLE
[HttpFoundation] Mark Request::get() internal

### LINKED ISSUE #40984: HttpFoundation: Deprecate Request::get
**Symfony version(s) affected**: all I think

**Description**

`Request::get` looks at the following data:

- `$this->attributes`
- `$this->query`
- `$this->request`

in that order. So it checks `$_GET` before it checks `$_POST`. I haven't checked, but I think it might have been implemented this way because of a misunderstanding about how the ini setting `request_order` works: The default there is `GP`, but

> Registration is done from left to right, newer values override older values.

so `P` (POST) takes precedence over `G` (GET)

and if you think about it, that is the only logical order, because the way Symfony does it, allows for trivial attacks by sending malicious links to users which completely override whatever they enter in a (POST) form.

**How to reproduce**

If the backend code uses (directly or indirectly) `$request->get()` and you have a POST form in the frontend, anyone can make a malicious link like `/order?selectedproduct=ultra-deluxe-edition` which completely overrides anything the user enters in a form, without them knowing. So for POST forms, `$request->get()` is outright dangerous.

If your frontend code uses a GET form, then your backend might as well just use  `$request->query->get()`, because where is the POST data going to come from? The only way to get POST data in this scenario would be if the backend code adds it itself programmatically. But if the backend code does that, it might as well just add its stuff to `$request->attributes` or modify `$request->query` directly.  

**Possible Solution**

There seems to be no scenario where `Request::get` can be used safely and does something useful. One way to fix this would be to invert the order, i.e. look at POST before looking at GET, but that would be a very hard to detect BC break. So it might be best to just get rid of the function completely

**Additional context**

See discussion in #40974

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | Fix #40984
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

we should not forget about this :)

### CODE REVIEW

**Comment by ro0NL** [comment]
yeah much internal usage.

no time to do such maintenance, sorry. Feel free to takeover.

**Comment by ro0NL** [comment]
`@internal` also works for me, i can do that quick :+1:

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -703,9 +703,13 @@ public static function getHttpMethodParameterOverride()
      * @param mixed $default The default value if the parameter key does not exist
      *
      * @return mixed
+     *
+     * @deprecated since Symfony 5.4
```
Comment (chalasr) [inline src/Symfony/Component/HttpFoundation/Request.php]: Providing the alternative would be helpful here, same for the notice

**Comment by ro0NL** [comment]
to make this PR actionable, i chosen `@internal` approach.

The goal to me is to get this method out of public API scope.

**Review by chalasr — APPROVED** [review]

**Comment by Nyholm** [comment]
Im okey removing it from the public API. In my opinion it should never be used by application authors.

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request merged**